tests/k6/scenarios/core2_qa_control_plane_ops.js

```diff
@@ -0,0 +1,206 @@
+                let replicasVariation = Math.round(Math.random() * deltaReplicas)
+                let newReplicas = model.spec.replicas + (replicasVariation * plusOrMinus)
+                if (newReplicas < 1) newReplicas = 1
+
+                let newModelCR = {
+                    "apiVersion": "mlops.seldon.io/v1alpha1",
+                    "kind": "Model",
+                    "metadata": {
+                        "name": modelName,
+                        "namespace": config.namespace
+                    },
+                    "spec": {
+                        "storageUri": model.spec.storageUri,
+                        "requirements": model.spec.requirements,
+                        "memory": newMemory,
+                        "replicas": newReplicas
+                    }
+                }
+                modelCRYaml = yamlDump(newModelCR)
+            } catch (_) {
+                // just continue test, another VU might have deleted the chosen model
+                return false
+            }
+            break;
+    }
+
+    console.log(`VU ${vu.idInTest} executes ${op.description} on ${modelName}`)
+
+
+    // execute control-plane operation
+    var opOk = false
+    switch(op) {
+        case seldonOpType.CREATE:
+        case seldonOpType.UPDATE:
+            opOk = k8s.loadModel(modelName, modelCRYaml, true)
+            break;
+        case seldonOpType.DELETE:
+            opOk = k8s.unloadModel(modelName, true)
+            break;
+    }
+
+    // update k6 metrics
+    if (opOk === true) {
+        switch(op) {
+            case seldonOpType.CREATE:
+                createCounter.add(1)
+                break;
+            case seldonOpType.UPDATE:
+                updateCounter.add(1)
+                break;
+            case seldonOpType.DELETE:
+                deleteCounter.add(1)
+                break;
+        }
+    }
+
+    return opOk
+
+}
+
+export default function (config) {
+    kubeclient = k8s.init()
+    const numModelTypes = config.modelType.length
+
+    var idx = Math.floor(Math.random() * numModelTypes)
+    while (config.maxNumModels[idx] === 0) {
+        idx = Math.floor(Math.random() * numModelTypes)
+    }
+    let modelsOfType = k8s.getModelsWithNamePrefix(config.modelNamePrefix[idx])
+
+    // Pick random operation in CREATE/UPDATE/DELETE, amongst available ones.
+    // Each operation is added multiple times in the selection array as
+    // configured in the `createUpdateDeleteBias` array. This defines the
+    // probability ratios between the operations. For example, the [1,4,3]
+    // createUpdateDeleteBias array makes an Update four times more likely then
+    // a Create, and a Delete 3 times more likely than the Create.
+    //
+    // Because each VU picks the operation independently, it's possible to
+    // temporarily get more models than MAX_NUM_MODELS for a given model type.
+    let availableOps = Array(config.createUpdateDeleteBias[1]).fill(seldonOpType.UPDATE)
```

Comment:
Can the list of ops contain zero items from a specific operation? Ie delete?

Reply:
Yes, any of the operations can be suppressed from the createUpdateDeleteBias array. It has three elements: `[a, b, c]`, where `a` corresponds to the "create bias", `b` corresponds to the "update bias" and `c` corresponds to the "delete bias". Setting one or some of them to 0 will completely supress the corresponding operation.